source/server/listener_manager_impl.h

```diff
@@ -166,6 +167,15 @@ class ListenerManagerImpl : public ListenerManager, Logger::Loggable<Logger::Id:
     stats_.total_listeners_warming_.set(warming_listeners_.size());
     stats_.total_listeners_active_.set(active_listeners_.size());
   }
+  bool listnerStopped(const envoy::api::v2::Listener& config) {
+    // Currently all listeners in a given direction are stopped because of the way admin
+    // drain_listener functionality is implemented. So it is OK to check traffic direction. This
+    // needs to be revisited, if that changes - if we support drain by listener name,for example.
+    return (config.traffic_direction() == envoy::api::v2::core::TrafficDirection::INBOUND &&
+            inbound_listeners_stopped_) ||
+           (config.traffic_direction() == envoy::api::v2::core::TrafficDirection::OUTBOUND &&
+            outbound_listeners_stopped_);
```

Comment:
Similarly, this requires traffic direction to be configured for `/drain_listeners` to work.